Mage/src/main/java/mage/abilities/effects/ContinuousEffectImpl.java

```diff
@@ -56,9 +56,11 @@ A Characteristic Defining Ability (CDA) is an ability that defines a characteris
 
     // until your next turn or until end of your next turn
     private UUID startingControllerId; // player to check for turn duration (can't different with real controller ability)
+    private UUID activePlayerId; // Player whose turn the effect started on
     private boolean startingTurnWasActive; // effect started during related players turn and related players turn was already active
     private int effectStartingOnTurn = 0; // turn the effect started
-    private int effectStartingEndStep = 0;
+    private int effectControllerStartingEndStep = 0;
+    private int effectActivePlayerStartingEndStep = 0;
```

Comment:
Use effecStartingXXX and startingActivePlayer naming

Reply:
Done.